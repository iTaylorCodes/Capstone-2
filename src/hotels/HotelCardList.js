import HotelCard from "./HotelCard";

/** Hotel Card List.
 *
 * Component dedicated to showing a list of HotelCard components.
 *
 * Renders multiple HotelCard components, passing in data from HotelsApi.
 *
 * Rendered by Neighborhood.
 */
const HotelCardList = ({ hotels }) => {
  return (
    <div className="row g-0 d-flex justify-content-center">
      {hotels.map((h) => {
        return (
          <div className="col-10 col-sm-6" key={h.id}>
            <HotelCard hotel={h} />
          </div>
        );
      })}
    </div>
  );
};

export default HotelCardList;
