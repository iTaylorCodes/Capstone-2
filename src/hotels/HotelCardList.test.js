import HotelCardList from "./HotelCardList";
import { render } from "@testing-library/react";

const hotels = [
  {
    id: 634418464,
    name: "The Grand NYC",
    starRating: 2,
    neighbourhood: "NoMad",
    address: {
      streetAddress: "38 West 31st Street",
      locality: "New York",
      postalCode: "10001",
      region: "NY",
      countryName: "United States",
    },
  },
  {
    id: 634418464,
    name: "The Grand NYC",
    starRating: 2,
    neighbourhood: "NoMad",
    address: {
      streetAddress: "38 West 31st Street",
      locality: "New York",
      postalCode: "10001",
      region: "NY",
      countryName: "United States",
    },
  },
];

it("renders without crashing", () => {
  render(<HotelCardList hotels={hotels} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<HotelCardList hotels={hotels} />);
  expect(asFragment()).toMatchSnapshot();
});
