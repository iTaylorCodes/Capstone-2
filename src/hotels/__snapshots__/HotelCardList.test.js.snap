// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row g-0 d-flex justify-content-center"
  >
    <div
      class="col-10 col-sm-6"
    >
      <div
        class="HotelCard m-2"
      >
        <div
          class="card"
          style="max-width: 435px;"
        >
          <img
            alt="The Grand NYC"
            src=""
            style="max-height: 300px;"
          />
          <h3
            class="card-header"
          >
            The Grand NYC
          </h3>
          <div
            class="card-body"
          >
            <h4
              class="card-title"
            >
              Rating: 
              <i
                class="fa fa-star"
                style="color: gold;"
              />
              <i
                class="fa fa-star"
                style="color: gold;"
              />
            </h4>
            <p
              class="card-text"
            >
              <span
                class="fw-bold"
              >
                Neighborhood: 
              </span>
              NoMad
            </p>
            <p
              class="card-text"
            >
              <span
                class="fw-bold"
              >
                Address: 
              </span>
              38 West 31st Street, New York, NY 10001
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-10 col-sm-6"
    >
      <div
        class="HotelCard m-2"
      >
        <div
          class="card"
          style="max-width: 435px;"
        >
          <img
            alt="The Grand NYC"
            src=""
            style="max-height: 300px;"
          />
          <h3
            class="card-header"
          >
            The Grand NYC
          </h3>
          <div
            class="card-body"
          >
            <h4
              class="card-title"
            >
              Rating: 
              <i
                class="fa fa-star"
                style="color: gold;"
              />
              <i
                class="fa fa-star"
                style="color: gold;"
              />
            </h4>
            <p
              class="card-text"
            >
              <span
                class="fw-bold"
              >
                Neighborhood: 
              </span>
              NoMad
            </p>
            <p
              class="card-text"
            >
              <span
                class="fw-bold"
              >
                Address: 
              </span>
              38 West 31st Street, New York, NY 10001
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
